src/internal/connector/onedrive/collections.go

```diff
@@ -451,17 +530,19 @@ func (c *Collections) UpdateCollections(
 				continue
 			}
 
-			// TODO(ashmrtn): Figure what when an item was moved (maybe) and add it to
-			// the exclude list.
-
 			col, found := c.CollectionMap[collectionPath.String()]
-
 			if !found {
-				// TODO(ashmrtn): Compare old and new path and set collection state
-				// accordingly.
+				var prevPath path.Path
+				// TODO(meain): better check for root
+				if strings.HasSuffix(collectionPath.Folder(), "root:") && !invalidPrevDelta {
+					// Root collection is always added with not moved flag
+					prevPath = collectionPath
+				}
```

Comment:
may be better to handle this by adding the root folder to the set of metadata we save. Then we don't need extra logic for every item

Reply:
I've added in a TODO there for now, let me pick it up in a follow up PR. It requires a bit more changes in the code.